sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -199,6 +209,19 @@ private[sql] case class FileSourceScanExec(
         options = relation.options,
         hadoopConf = relation.sparkSession.sessionState.newHadoopConfWithOptions(relation.options))
 
+      (file: PartitionedFile) => {
+        val iter = func(file)
+        // Only for test purpose.
+        // Once the vectorized Parquet reader is initialized in the above method, we can read its
+        // variable numRowGroups.
+        if (fileFormat != null) {
```

Comment:
hmm, VectorizedParquetRecordReader is not exposed to outside of ParquetFileFormat. It is wrapped in a returned anonymous function that takes a PartitionedFile and read data from it. In order to pass it to VectorizedParquetRecordReader, we might need to change current API of FileFormat. Is it worth doing that for this?